### PR TITLE
fix: improve `cutler rebase` ouput

### PR DESCRIPTION
**Pull request by eseidel**
Wasn't prefixing the dev branch with origin when
pushing to the release branches, so would sometimes
push a stale local branch.

Also needed to always output push commands even
when we aren't changing what shorebird/dev points to
just to update the tags for the repo, as well as pushing
the release branch.

Also added a couple simple tests to in part improve
coverage, but mostly to make later attempts to add more
testing easier.

Also imported package:io with an alias to not collide with
package:io exports from mason_logger which we now
export from logger.dart.

### CODE REVIEW

File: cutler/lib/commands/rebase_command.dart [hunk 1]
```diff
@@ -56,17 +61,38 @@ class RebaseContext {
...
     // Only need to push if we did a rebase (or we changed a dependency).
-    if (upstream[repo].ref != forkpoint[repo].ref) {
+    if (upstream[repo] != forkpoint[repo]) {
       return true;
     }
     return repo.dependencies.any(needsPush);
   }
 
+  /// Print the commands needed to push the given repo.
+  void printPushCommands(Repo repo, String flutterVersionName) {
```
Comment (bryanoltman) [inline cutler/lib/commands/rebase_command.dart:78]: Named parameters here

File: cutler/lib/commands/rebase_command.dart [hunk 2]
```diff
@@ -76,15 +102,33 @@ class RebaseContext {
...
-      logger.warn('Need to update dart version.');
+      logger
+        ..warn('Need to update dart_sdk_revision version in $engineDeps')
+        ..info('git -C ${Repo.dart.path} rev-parse HEAD');
     }
+    if (needsPush(Repo.engine)) {
+      logger
+        ..warn('Once DEPS are updated above, need to update engine version')
+        ..info(
+          'git -C ${Repo.engine.path} commit -a -m '
```
Comment (bryanoltman) [inline cutler/lib/commands/rebase_command.dart:124]: Ultranit: `-m -a` can be shortened to `-am`

File: cutler/lib/commands/rebase_command.dart [hunk 3]
```diff
@@ -76,15 +102,33 @@ class RebaseContext {
...
     if (needsPush(Repo.dart)) {
       // If this changes it will inevitably have conflicts to resolve anyway.
-      logger.warn('Need to update dart version.');
+      logger
+        ..warn('Need to update dart_sdk_revision version in $engineDeps')
+        ..info('git -C ${Repo.dart.path} rev-parse HEAD');
     }
+    if (needsPush(Repo.engine)) {
+      logger
+        ..warn('Once DEPS are updated above, need to update engine version')
```
Comment (bryanoltman) [inline cutler/lib/commands/rebase_command.dart:122]:
```suggestion
        ..warn("Once DEPS are updated above, need to update the flutter repo's bin/internal/engine.version")
```

**Review by bryanoltman — APPROVED** [review]

File: cutler/lib/commands/rebase_command.dart (hunk 3 above)
Comment (eseidel) [reply cutler/lib/commands/rebase_command.dart]: Done.

File: cutler/lib/commands/rebase_command.dart (hunk 2 above)
Comment (eseidel) [reply cutler/lib/commands/rebase_command.dart]: Done.

**Pull request merged**